test/setup.ts

```diff
@@ -0,0 +1,50 @@
+import 'core-js/proposals/reflect-metadata'
+import { NydusClient, NydusServer } from 'nydus'
+import { container, instanceCachingFactory } from 'tsyringe'
+import { UpdateOrInsertUserIp } from '../server/lib/network/user-ips-type'
+import { RequestSessionLookup, SessionInfo } from '../server/lib/websockets/session-lookup'
+import {
+  ClientSocketsGroup,
+  ClientSocketsManager,
+  UserSocketsGroup,
+  UserSocketsManager,
+} from '../server/lib/websockets/socket-groups'
+
+export class FakeNydusServer extends NydusServer {
```

Comment:
I think I'd probably fake this whole thing, stubbing out particular methods only seems very dangerous for updates. I *think* you can do something like:

```ts
class FakeNydusServer extends EventEmitter implements NydusServer {
  // ...
}
```

to get the EventEmitter methods implemented for you (those are fine to leave unfaked imo) and then either include empty methods for the rest of the API, or implement things to help with testing (logging events to lists to verify later, etc.)